1.15.2 - NullPointerException onCombat() Event

Hi! First thanks to your light and simple plugin ! I tried it on my private server (Spigot 1.15.2 lastest version). This error appear when player attack all entities or get damage by all entities.

I use the 4.2.0 version of your plugin.

> [12:40:35] [Server thread/ERROR]: Could not pass event EntityDamageByEntityEvent to AntiRelog v4.2.0
[...]
Caused by: java.lang.NullPointerException
	at pl.jeremi.antirelog.AntiRelog.onCombat(AntiRelog.java:90) ~[?:?]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_241]
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source) ~[?:1.8.0_241]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source) ~[?:1.8.0_241]
	at java.lang.reflect.Method.invoke(Unknown Source) ~[?:1.8.0_241]
	at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:315) ~[spigot-1.15.2.jar:git-Spigot-6de3d4b-a6f8093]
	... 28 more

--- a/src/main/java/pl/jeremi/antirelog/AntiRelog.java
+++ b/src/main/java/pl/jeremi/antirelog/AntiRelog.java
@@ -84,8 +84,11 @@ public final class AntiRelog extends JavaPlugin implements Listener {
         return false;
     }
 
-    @EventHandler
+    @EventHandler(priority = EventPriority.HIGHEST)
     public void onCombat(final EntityDamageByEntityEvent event) {
+        if(event.isCancelled())
+            return;
+
         if (event.getDamager() instanceof Player && isSubject(event.getEntity().getType())) {
             Player player = (Player) event.getDamager();
             if (!bypassingPlayers.get(player))

--- a/src/main/java/pl/jeremi/antirelog/AntiRelog.java
+++ b/src/main/java/pl/jeremi/antirelog/AntiRelog.java
@@ -15,9 +15,9 @@ import org.bukkit.event.entity.PlayerDeathEvent;
 import org.bukkit.event.player.PlayerJoinEvent;
 import org.bukkit.event.player.PlayerQuitEvent;
 import org.bukkit.plugin.java.JavaPlugin;
+import org.jetbrains.annotations.NotNull;
 
 import java.util.HashMap;
-import java.util.logging.Level;
 
 /**
  * Created by Jeremiasz N. on 2016-04-26.
@@ -25,7 +25,6 @@ import java.util.logging.Level;
 public final class AntiRelog extends JavaPlugin implements Listener {
     static FileConfiguration config;
     private final HashMap<Player, CombatHandle> handledPlayers = new HashMap<>();
-    private final HashMap<Player, Boolean> bypassingPlayers = new HashMap<>();
 
     @Override
     public void onEnable() {
@@ -53,32 +52,9 @@ public final class AntiRelog extends JavaPlugin implements Listener {
     }
 
     @Override
-    public boolean onCommand(CommandSender sender, Command command, String label, String[] args) {
-        if (command.getName().equalsIgnoreCase("artoggle")) {
-            if (sender instanceof Player && args.length == 0) {
-                Player player = (Player) sender;
-                if (bypassingPlayers.containsKey(player)) {
-                    bypassingPlayers.put(player, !(bypassingPlayers.get(player)));
-                    getLogger().log(Level.INFO, "Toggled " + player.getName() + "'s bypass to " + bypassingPlayers.get(player).toString());
-                    player.sendMessage("[AntiRelog] " + ChatColor.GREEN + "Successfully set your bypass: " + bypassingPlayers.get(player).toString());
-                }
-                return true;
-            } else if (args.length == 1) {
-                Player player = getServer().getPlayer(args[0]);
-                if (player != null && sender instanceof Player && !player.equals(sender) && !sender.hasPermission("antirelog.toggle.others")) {
-                    sender.sendMessage("[AntiRelog] " + ChatColor.RED + "You don't have permission to toggle others bypass.");
-                    return true;
-                }
-                if (player != null && bypassingPlayers.containsKey(player)) {
-                    bypassingPlayers.put(player, !(bypassingPlayers.get(player)));
-                    getLogger().log(Level.INFO, "Toggled " + player.getName() + "'s bypass to " + bypassingPlayers.get(player).toString());
-                    player.sendMessage("[AntiRelog] " + ChatColor.GREEN + "Successfully set your bypass: " + bypassingPlayers.get(player).toString());
-                } else {
-                    sender.sendMessage("[AntiRelog] " + ChatColor.RED + "Something went wrong. Is this player online?");
-                }
-                return true;
-            }
-        } else if (command.getName().equalsIgnoreCase("arreload")) {
+    public boolean onCommand(@NotNull CommandSender sender, Command command, @NotNull String label, String[] args) {
+        if (command.getName().equalsIgnoreCase("arreload")) {
+            sender.sendMessage("[AntiRelog] " + ChatColor.GREEN + "Config was successfully reloaded.");
             reloadConfig();
         }
         return false;
@@ -90,32 +66,23 @@ public final class AntiRelog extends JavaPlugin implements Listener {
             return;
 
         if (event.getDamager() instanceof Player && isSubject(event.getEntity().getType())) {
-            Player player = (Player) event.getDamager();
-            if (!bypassingPlayers.get(player))
+            final Player player = (Player) event.getDamager();
+            if (!player.hasPermission("antirelog.bypass"))
                 handledPlayers.get(player).startCombat();
-        }
+        } else if (event.getDamager() instanceof Projectile) {
+            if (((Projectile) event.getDamager()).getShooter() instanceof Player && isSubject(event.getEntity().getType())) {
+                final Player damager = (Player) (((Projectile) event.getDamager()).getShooter());
 
-        if (event.getEntity() instanceof Player) {
-            Player player = (Player) event.getEntity();
-            if (!bypassingPlayers.get(player))
-                handledPlayers.get(player).startCombat();
-        }
-
-        if (event.getDamager() instanceof Projectile) {
-            if (((Projectile) event.getDamager()).getShooter() instanceof Player &&
-                    isSubject(event.getEntity().getType())) {
-                Player damager = (Player) (((Projectile) event.getDamager()).getShooter());
-                if (!bypassingPlayers.get(damager))
+                if (damager != null && !damager.hasPermission("antirelog.bypass"))
                     handledPlayers.get(damager).startCombat();
             }
         }
-    }
 
-    protected static String getConfigString(String key) {
-        String value = config.getString(key);
-        return value != null
-                ? value
-                : "";
+        if (event.getEntity() instanceof Player) {
+            final Player player = (Player) event.getEntity();
+            if (!player.hasPermission("antirelog.bypass"))
+                handledPlayers.get(player).startCombat();
+        }
     }
 
     private boolean isSubject(EntityType entity) {
@@ -127,16 +94,16 @@ public final class AntiRelog extends JavaPlugin implements Listener {
 
     @EventHandler
     public void onJoin(final PlayerJoinEvent event) {
-        handledPlayers.put(event.getPlayer(), new CombatHandle(event.getPlayer(), this));
-        bypassingPlayers.put(event.getPlayer(), event.getPlayer().hasPermission("antirelog.bypass"));
+        final Player player = event.getPlayer();
+        handledPlayers.put(player, new CombatHandle(player, this));
     }
 
     @EventHandler(priority = EventPriority.HIGHEST)
     public void onQuit(final PlayerQuitEvent event) {
-        Player player = event.getPlayer();
-        if (!bypassingPlayers.get(player) && handledPlayers.get(player).isInCombat()) {
+        final Player player = event.getPlayer();
+        if (!player.hasPermission("antirelog.bypass") && handledPlayers.get(player).isInCombat()) {
             player.setHealth(0);
-            String broadcastMessage = config.getString("broadcast-message");
+            final String broadcastMessage = config.getString("broadcast-message");
             if (broadcastMessage != null && !broadcastMessage.isEmpty()) {
                 event.setQuitMessage(ChatColor.translateAlternateColorCodes(
                         '&',
@@ -150,13 +117,11 @@ public final class AntiRelog extends JavaPlugin implements Listener {
             handledPlayers.get(player).cleanUp();
             handledPlayers.remove(player);
         }
-        bypassingPlayers.remove(player);
     }
 
     @EventHandler
     public void onDeath(final PlayerDeathEvent event){
-        Player player = event.getEntity().getPlayer();
-        CombatHandle combatHandle = handledPlayers.get(player);
+        final CombatHandle combatHandle = handledPlayers.get(event.getEntity().getPlayer());
         if(combatHandle.isInCombat()){
             combatHandle.reset();
         }

--- a/src/main/java/pl/jeremi/antirelog/AntiRelog.java
+++ b/src/main/java/pl/jeremi/antirelog/AntiRelog.java
@@ -24,8 +24,8 @@ import java.util.logging.Level;
  */
 public final class AntiRelog extends JavaPlugin implements Listener {
     static FileConfiguration config;
-    private HashMap<Player, CombatHandle> handledPlayers = new HashMap<>();
-    private HashMap<Player, Boolean> bypassingPlayers = new HashMap<>();
+    private final HashMap<Player, CombatHandle> handledPlayers = new HashMap<>();
+    private final HashMap<Player, Boolean> bypassingPlayers = new HashMap<>();
 
     @Override
     public void onEnable() {
@@ -152,6 +152,7 @@ public final class AntiRelog extends JavaPlugin implements Listener {
         }
         bypassingPlayers.remove(player);
     }
+
     @EventHandler
     public void onDeath(final PlayerDeathEvent event){
         Player player = event.getEntity().getPlayer();

--- a/src/main/java/pl/jeremi/antirelog/CombatHandle.java
+++ b/src/main/java/pl/jeremi/antirelog/CombatHandle.java
@@ -32,14 +32,14 @@ class CombatHandle {
 
         if (enableBar) {
             busyBar = Bukkit.createBossBar(formatCombatChatMessage("busy-message"),
-                    BarColor.valueOf(AntiRelog.getConfigString("busy-color").toUpperCase()),
-                    BarStyle.valueOf(AntiRelog.getConfigString("busy-style").toUpperCase()));
+                    BarColor.valueOf(AntiRelog.config.getString("busy-color").toUpperCase()),
+                    BarStyle.valueOf(AntiRelog.config.getString("busy-style").toUpperCase()));
             busyBar.addPlayer(player);
             busyBar.setVisible(false);
 
             freeBar = Bukkit.createBossBar(formatCombatChatMessage("free-message"),
-                    BarColor.valueOf(AntiRelog.getConfigString("free-color").toUpperCase()),
-                    BarStyle.valueOf(AntiRelog.getConfigString("free-style").toUpperCase()));
+                    BarColor.valueOf(AntiRelog.config.getString("free-color").toUpperCase()),
+                    BarStyle.valueOf(AntiRelog.config.getString("free-style").toUpperCase()));
             freeBar.addPlayer(player);
             freeBar.setVisible(false);
         }
@@ -48,7 +48,7 @@ class CombatHandle {
     }
 
     private String formatCombatChatMessage(String message) {
-        return ChatColor.translateAlternateColorCodes('&', AntiRelog.getConfigString(message))
+        return ChatColor.translateAlternateColorCodes('&', AntiRelog.config.getString(message))
                 .replaceAll("\\{displayname}", player.getDisplayName())
                 .replaceAll("\\{username}", player.getName())
                 .replaceAll("\\{timeleft}", String.valueOf(combatTimeLeft))

--- a/src/main/resources/plugin.yml
+++ b/src/main/resources/plugin.yml
@@ -10,30 +10,13 @@ permissions:
     children:
       antirelog.reload: true
       antirelog.bypass: true
-      antirelog.toggle: true
-      antirelog.toggle.others: true
   antirelog.reload:
     description: Reload Anti-Relog config.
     default: op
   antirelog.bypass:
     description: Bypass Anti-Relog protection.
     default: op
-  antirelog.toggle:
-    description: Toggle AntiRelog bypass.
-    default: op
-  antirelog.toggle.others:
-    description: Toggle AntiRelog bypass for other players.
-    default: op
-    children:
-      antirelog.toggle: true
 commands:
-  artoggle:
-    description: AntiRelog main command
-    permission: antirelog.toggle
-    permission-message: "[AntiRelog] You don't have permission to do that!"
-    usage: |
-      AntiRelog by Jeremi
-      /artoggle [player]
   arreload:
     description: AntiRelog reload config
     permission: antirelog.reload

